ql/src/test/org/apache/hadoop/hive/ql/lockmgr/TestDbTxnManager.java

```diff
@@ -516,10 +516,17 @@ public void testHeartbeaterReplicationTxn() throws Exception {
     } catch (LockException e) {
       exception = e;
     }
-    Assert.assertNotNull("Txn should have been aborted", exception);
-    Assert.assertEquals(ErrorMsg.TXN_ABORTED, exception.getCanonicalErrorMsg());
+    Assert.assertNotNull("Source transaction with txnId: 1, missing from REPL_TXN_MAP", exception);
```

Comment:
The message you get if the assertion fails. If this entry is missing, the exception object would be null. Is that expected?

Reply:
If this entry is missing, the exception would be thrown which will be caught up in line 517. So, e would be not null.